Automattic/jetpack
# Forms: refactor Multiple and Single Choice fields

## Proposed changes:

https://github.com/Automattic/jetpack/pull/38916 made one of the Calyso e2e tests fail (see conversion linked below) because it changed the markup of the _Multiple Choice_ and _Single Choice_ blocks in the editor. It highlighted that [the block and the inner block props](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/) were passed incorrectly. This PR refactors these two blocks to match how Gutenberg handles block and inner blocks (e.g. [the List block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/list)). It should fix the failing e2e test as well.

In detail, the PR does the following:
- Move out the code for _Multiple Choice_ and _Single Choice_ out of `child-blocks.js` for readability, respectively in the `jetpack-field-multiple-choice` and `jetpack-field-single-choice` folders
- Extract the code shared by these two fields in `jetpack-field-choice`
- Updated some selectors in `editor.scss` to prevent any design breakage
- Extract `getFieldLabel` in a separate file
- Create the `mergeSettings` utility to merge a block settings with default settings
- Deleted the now unused `jetpack-field-option.js` file

### Other information:


## Jetpack product discussion


Conversation: p1724766845302749-slack-C01U2KGS2PQ

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with the `trunk` branch. You need to be able to switch branch later on.
- Create a new post
- Add a _Multiple Choice_ field from the block inserter (it will be wrapped in a form)
<img width="347" alt="Screenshot 2024-09-03 at 2 59 12 PM" src="https://github.com/user-attachments/assets/227ff81b-9f9b-4d7e-8a7a-5509d5186842">


- Add a _Single Choice_ field to the form
<img width="346" alt="Screenshot 2024-09-03 at 2 59 21 PM" src="https://github.com/user-attachments/assets/c68cd5fa-99aa-457f-ab4f-9030daea9cc6">


- Save the draft and open the preview in a new tab
- In a new tab, update the Jetpack version so it uses this branch
- Visit the post you've created. You should not see any errors in the UI or the console.
- Verify the blocks work like they do on trunk
- Open the preview and make sure the form works as expected

_Note: the label is different in the editor and the preview as long as it's not edited. This issue is already present on trunk and will be addressed in a separate PR._
